weldx/measurement.py

```diff
@@ -53,55 +60,507 @@ class DataTransformation:
+            unit=output_signal_unit,
+            signal=output_signal,
+        )
+        if signal_data is not None:
+            self.add_signal_data(signal_data)  # todo : test
 
     @staticmethod
-    def _add_node(
-        node: str,
-        parent_node: str,
-        node_label: str,
-        position: Tuple[float, float],
-        container: Tuple["DiGraph", List, Dict, Dict],
-    ):  # pragma: no cover
-        """Add a new node to several containers.
+    def construct_from_source(name, source: SignalSource):
+
+        return MeasurementChain(
+            name=name,
+            source_name=source.name,
+            source_error=source.error,
+            output_signal_type=None,
+            output_signal_unit=None,
+            output_signal=source.output_signal,
+            signal_data=None,
+        )
+
+    @staticmethod
+    def construct_from_equipment(name, equipment: GenericEquipment, source_name=None):
+        if len(equipment.sources) > 1:
+            if source_name is None:
+                raise ValueError(
+                    "The equipment has multiple sources. Specify the "
+                    "desired one by providing a 'source_name'."
+                )
+            source = equipment.get_source(source_name)
+        elif len(equipment.sources) == 1:
+            source = equipment.sources[0]
+        else:
+            raise ValueError("The equipment does not have a source.")
+
+        mc = MeasurementChain.construct_from_source(name, source)
+        mc._source["equipment"] = equipment
+        return mc
+
+    @staticmethod
+    def construct_from_tree(tree: Dict) -> "MeasurementChain":
+        mc = MeasurementChain(
+            name=tree["name"],
+            source_name="source",
+            source_error=Error(1),
+            output_signal_type="analog",
+            output_signal_unit="V",
+        )
+        # todo: implement correct version, when schema is ready
+        return mc
 
-        This is a helper for the plot function.
+    def _add_signal(self, node_id: str, signal_type: str, unit: str, signal: Signal):
+        """Add a new signal node to internal graph.
+
+        Parameters
+        ----------
+        node_id :
+            The name that will be used for the internal graphs' node. This is identical
+            to the name of the signals source.
+        signal_type :
+            Type of the signal (analog or digital)
+        unit :
+            Unit of the signal
+
+        """
+        self._raise_if_node_exist(node_id)
+        signal = self._create_signal(signal_type, unit, signal)
+
+        self._graph.add_node(node_id, signal=signal)
+        self._prev_added_signal = node_id
+
+    def _raise_if_node_exist(self, node_id: str):
+        """Raise en error if the graph already contains a node with the passed id.
+
+        Parameters
+        ----------
+        node_id :
+            Name that should be checked
+
+        """
```

Comment:
```suggestion
        """Raise en error if the graph already contains a node with the passed id."""
```
A short description instead of a full docstring feels more readable for internal functions like this that are very descriptive with type-hints